lib/parse.js

```diff
@@ -2109,7 +2109,7 @@ function parse($TEXT, options) {
         return function_(AST_Accessor, is_generator, is_async);
     });
 
-    var object_or_object_destructuring_ = embed_tokens(function() {
+    var object_or_destructuring_ = embed_tokens(function object_or_destructuring_() {
```

Comment:
What I meant is this:
```js
var foo = embed_tokens(funciton foo() {
  doA();
});
doB(foo());
```
If later on I need to move the logic around such that:
```js
var foo =  embed_tokens(funciton foo() {
  doA();
  doB(foo());
});
```
I now inadvertently introduced a bug because `foo` is now referring to `function foo(...` as opposed to `var foo = embed_tokens(...`